File: ai-hub-backend/src/models/adminModel.js
import mongoose from 'mongoose';

const adminSchema = new mongoose.Schema({
  email: {
    type: String,
    required: true,
    unique: true
  },
  name: {
    type: String,
  },
  isVerified: {
    type: Boolean,
    default: false
  },
  password: {
    type: String,
    required: true
  },
  role: {
    type: String,
    default: 'admin'
  },
  verificationCode: {
    type: String,
    default: null
  },
  verificationCodeExpires: {
    type: Date,
    default: null
  },
  isResetCodeConfirmed: {
    type: Boolean,
    default: false
  }
}, { timestamps: true });

export default mongoose.model('Admin', adminSchema);

File: backend/infra/db/mongoose/models/admin_model.go
package models

import "time"

type AdminDoc struct {
	ID                      string     `bson:"_id"`
	Email                   string     `bson:"email"`
	Name                    *string    `bson:"name,omitempty"`
	IsVerified              bool       `bson:"isVerified"`
	Password                string     `bson:"password"`
	Role                    string     `bson:"role"`
	VerificationCode        *string    `bson:"verificationCode,omitempty"`
	VerificationCodeExpires *time.Time `bson:"verificationCodeExpires,omitempty"`
	IsResetCodeConfirmed    bool       `bson:"isResetCodeConfirmed"`
	CreatedAt               time.Time  `bson:"createdAt"`
	UpdatedAt               time.Time  `bson:"updatedAt"`
}
